### PR TITLE
Dev/serializer updates

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
- Adding std::vector constructor and copy assignment operator for the bit array class.
- Adding set_bit_array() function to SerializableStateFields, and increasing test coverage of such state fields.

### CODE REVIEW

File: src/fixed_array.hpp [hunk 1]
```diff
@@ -63,6 +74,20 @@ class fixed_array_base : public std::vector<T> {
...
+    /**
+     * @brief Allows assignment-by-value using an STL vector. If the arrays are not of the
+     * same length, nothing happens.
+     *
+     * @tparam sz
+     * @param set
+     * @return fixed_array&
+     */
+    fixed_array_base& operator=(const std::vector<T>& arr) {
+        if (arr.size() != this->size()) return *this;
```
Comment (athena255) [inline src/fixed_array.hpp]: Why return a pointer to ourselves when the return type is a reference?

File: test/bit_array/test_bit_array.cpp [hunk 2]
```diff
@@ -17,6 +17,13 @@ void test_bitarray_constructors() {
     bit_array arr2(*arr);
     TEST_ASSERT_EQUAL(10, arr2.size());
     TEST_ASSERT_EQUAL(1, arr2[0]);
+
+    // Copy constructor from std::vector (also tests assignment operator)
+    std::vector<bool> arr3(8);
+    arr3[5] = 1;
+    bit_array arr4(arr3);
+    TEST_ASSERT_EQUAL(8, arr4.size());
+    TEST_ASSERT_EQUAL(1, arr4[5]);
```
Comment (athena255) [inline test/bit_array/test_bit_array.cpp]: might be good to also test that they are not  the same object

File: src/fixed_array.hpp (hunk 1 above)
Comment (tanishqaggarwal) [reply src/fixed_array.hpp]: `this` is a pointer. `*this` is the dereferenced value of the pointer, i.e. a reference to the current object.

File: test/bit_array/test_bit_array.cpp (hunk 2 above)
Comment (tanishqaggarwal) [reply test/bit_array/test_bit_array.cpp]: Yeah good point

**Pull request merged**